src/syntax/Scanner.js

```diff
@@ -527,17 +527,31 @@ function peekTokenNoLineTerminator() {
   let end = t.location.start.offset;
   for (let i = start; i < end; i++) {
     let code = input.charCodeAt(i);
-    if (isLineTerminator(code))
+    if (isLineTerminator(code)) {
       return null;
+    }
 
-    // If we have a block comment we need to skip it since new lines inside
-    // the comment are not significant.
+    // If a block comment contains new lines then the comment is treated as a
+    // new line.
     if (code === 47) {  // '/'
       code = input.charCodeAt(++i);
       // End of line comments always mean a new line is present.
-      if (code === 47)  // '/'
+      if (code === 47) {  // '/'
         return null;
-      i = input.indexOf('*/', i) + 2;
+      }
+
```

Comment:
I thought the same thing at first. The reason why we don't need to check this is that we are only checking the text between the last token and the next token. That means that the text is either whitespace, a comment or a sequence of these.

I'll add a comment.